### PR TITLE
Removing temp files that are being created in extensions flow.

### PR DESCRIPTION
**Pull request by arisettisanjana**
So currently the present code at https://github.com/Azure/azure-linux-extensions/blob/20a56bd98a63953561a3066a0b58fba615e705d9/VMBackup/main/Utils/HandlerUtil.py#:~:text=f%20%3D%20tempfile.NamedTemporaryFile(delete%3DFalse) temporary files are being created to read the protected settings, once they have been read we are not removing or cleaning them.
In general they are cleaned up when reboot happens or the customer's VM cleans up the /tmp files every 10 days.
To improve this we are adding the logic to clean/remove them as soon as they read the protected settings.
Changes made in VMBackup/main/Utils/HandlerUtil.py

### CODE REVIEW

File: VMBackup/main/Utils/HandlerUtil.py [hunk 1]
```diff
@@ -221,6 +221,12 @@ def _parse_config(self, ctxt):
                     self.error('JSON exception decoding ' + cleartxt)
                 handlerSettings['protectedSettings'] = jctxt
                 self.log('Config decoded correctly.')
+                # cleaning/removing the temp files created
+                try:
+                    if os.path.isfile(f.name):
+                        os.remove(f.name)
+                except Exception as e:
+                    self.error('Failed to remove the temporary file that was created for reading the protected settings' + e)
```
Comment (iamashwani-who) [inline VMBackup/main/Utils/HandlerUtil.py:229]: We can limit the error message to ('Failed to remove the temporary file ' + e)

File: VMBackup/main/Utils/HandlerUtil.py [hunk 2]
```diff
@@ -221,6 +221,12 @@ def _parse_config(self, ctxt):
                     self.error('JSON exception decoding ' + cleartxt)
                 handlerSettings['protectedSettings'] = jctxt
                 self.log('Config decoded correctly.')
+                # cleaning/removing the temp files created
+                try:
+                    if os.path.isfile(f.name):
+                        os.remove(f.name)
```
Comment (iamashwani-who) [inline VMBackup/main/Utils/HandlerUtil.py:227]: it should be path than file name, correct?

Comment (arisettisanjana) [reply VMBackup/main/Utils/HandlerUtil.py:227]: the name attribute stores the absolute path.

**Review by iamashwani-who — APPROVED** [review]

File: VMBackup/main/Utils/HandlerUtil.py (hunk 2 above)
Comment (ankitha-p) [inline VMBackup/main/Utils/HandlerUtil.py:227]: are we sure this file is not referenced anywhere else during the execution cycle of the code.

Comment (arisettisanjana) [reply VMBackup/main/Utils/HandlerUtil.py:227]: yes

**Review by rakrashba — APPROVED** [review]

**Pull request merged**